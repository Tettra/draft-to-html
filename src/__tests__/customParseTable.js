import { convertFromRaw } from 'draft-js'
import convertToHtml from '../'
import table from './table.json'

const parseBlock = (block, contentState) => {
  if (block.getType().toLowerCase() === 'atomic') {
    const entity = contentState.getEntity(block.getEntityAt(0))
    if (entity != null && entity.getType() === 'TABLE') {
      const { rows } = entity.getData()
      const tableEl = document.createElement('table')
      const tbodyEl = document.createElement('tbody')

      tableEl.appendChild(tbodyEl)

      rows.forEach(row => {
        const rowEl = document.createElement('tr')
        row.forEach(cell => {
          const tdEl = document.createElement('td')
          tdEl.innerHTML = convertToHtml(convertFromRaw(cell))
          rowEl.appendChild(tdEl)
        })
        tbodyEl.appendChild(rowEl)
      })

      return tableEl
    }
  }
}

describe('convertToHtml with custom parseBlock option', () => {
  test('converts table', () => {
    const html = convertToHtml(convertFromRaw(table), { parseBlock })
    expect(html).toMatchSnapshot()
  })
})
