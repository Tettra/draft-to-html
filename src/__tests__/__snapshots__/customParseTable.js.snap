// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertToHtml with custom parseBlock option converts table 1`] = `"<table></table>"`;
